src/kudu/tserver/simple_tablet_manager.cc

```diff
@@ -350,12 +331,47 @@ Status TSTabletManager::SetupRaft() {
   // before unlocking.
   RETURN_NOT_OK(consensus_->Start(
         bootstrap_info, std::move(peer_proxy_factory),
-        log, std::move(time_manager),
+        log_, std::move(time_manager),
         this, server_->metric_entity(), mark_dirty_clbk_));
 
+  RETURN_NOT_OK_PREPEND(WaitUntilRunning(),
+                        "Failed waiting for the raft to run");
+
+  set_state(MANAGER_RUNNING);
   return Status::OK();
 }
 
+Status TSTabletManager::SetupRaft() {
+  CHECK_EQ(state(), MANAGER_INITIALIZING);
+
+  InitLocalRaftPeerPB();
+
+  ConsensusOptions options;
+  options.tablet_id = kSysCatalogTabletId;
+  shared_ptr<RaftConsensus> consensus;
+  TRACE("Creating consensus");
+  LOG(INFO) << LogPrefix(kSysCatalogTabletId) << "Creating Raft for the system tablet";
+  RETURN_NOT_OK(RaftConsensus::Create(std::move(options),
+                                      local_peer_pb_,
+                                      cmeta_manager_,
+                                      server_->raft_pool(),
+                                      &consensus));
+  consensus_ = std::move(consensus);
+  // set_state(INITIALIZED);
+  // SetStatusMessage("Initialized. Waiting to start...");
+
+  // Not sure these 2 lines are required
+  scoped_refptr<ConsensusMetadata> cmeta;
+  Status s = cmeta_manager_->Load(kSysCatalogTabletId, &cmeta);
+
+  // Open the log, while passing in the factory class.
+  // Factory could be empty.
+  LogOptions log_options;
+  log_options.log_factory = server_->opts().log_factory;
```

Comment:
Where is log factory initialized in the flow? Is there a diff that is going to come on top of this stack?

Reply:
log_factory is either empty or passed in by application, in our case MySQL plugin. if the plugin is uninitialized it falls back to default Log::Log()